app/lib/formatter.rb

```diff
@@ -259,8 +259,9 @@ def link_to_url(entity, options = {})
 
   def link_to_mention(entity, linkable_accounts)
     acct = entity[:screen_name]
+    domain = acct.split('@')[1]
 
-    return link_to_account(acct) unless linkable_accounts
+    return link_to_account(acct) unless linkable_accounts and domain != 'twitter.com'
```

Comment:
Since you're already splitting the domain here, maybe you can do something like `return link_to_twitter(username) if domain == 'twitter.com'` before this line?